File: Android/FoodRescue/app/src/main/java/edu/ewu/team1/foodrescue/fragments/EaterFragment.java
package edu.ewu.team1.foodrescue.fragments;

import android.content.Context;
import android.content.SharedPreferences;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ListView;
import android.widget.Switch;

import java.util.ArrayList;
import java.util.Collections;

import edu.ewu.team1.foodrescue.FoodEvent;
import edu.ewu.team1.foodrescue.FoodEventAdapter;
import edu.ewu.team1.foodrescue.R;

/**
 * A simple {@link Fragment} subclass.
 */
public class EaterFragment extends Fragment {

    public EaterFragment() {
        // Required empty public constructor
    }

    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        View view = inflater.inflate(R.layout.fragment_eater, container, false);
        SharedPreferences sharedPref = view.getContext().getSharedPreferences(getString(R.string.preference_file_key), Context.MODE_PRIVATE);

        //Set the state of the switches to the last state of the switch. If first time, set to true
        //Also, register a click listener to save the new state of the switch
        boolean state = sharedPref.getBoolean("receiveEventStartNotifications", false);
        Switch eStart = view.findViewById(R.id.switchEventStart);
        eStart.setChecked(state);
        eStart.setOnClickListener(v -> {
            SharedPreferences.Editor editor = sharedPref.edit();
            editor.putBoolean("receiveEventStartNotifications", eStart.isChecked());
            editor.apply();
        });

        String[] data = sharedPref.getString("foodEvents", "").split("\n");
        ArrayList<FoodEvent> events = new ArrayList<>();
        for (String d : data) {
            if (!d.isEmpty())
                events.add(new FoodEvent(d));
        }
        Collections.sort(events);
        ListView list = view.findViewById(R.id.listViewFoodEvents);
        list.setAdapter(new FoodEventAdapter(events, view.getContext()));

        return view;
    }

}

File: Android/FoodRescue/app/src/main/java/edu/ewu/team1/foodrescue/fragments/EaterFragment.kt
package edu.ewu.team1.foodrescue.fragments

import android.os.Bundle
import android.support.v4.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ListView
import android.widget.Switch
import edu.ewu.team1.foodrescue.MainActivity
import edu.ewu.team1.foodrescue.R
import edu.ewu.team1.foodrescue.utilities.DataManager
import edu.ewu.team1.foodrescue.utilities.FoodEvent
import edu.ewu.team1.foodrescue.utilities.FoodEventAdapter
import java.util.*

class EaterFragment : Fragment() {
	private lateinit var dataManager: DataManager

	override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
	                          savedInstanceState: Bundle?): View? {
		super.onCreateView(inflater, container, savedInstanceState)
		// Inflate the layout for this fragment
		val view = inflater.inflate(R.layout.fragment_eater, container, false)
		dataManager = (activity as MainActivity).dataManager
		//Set the state of the switches to the last state of the switch. If first time, set to true
		//Also, register a click listener to save the new state of the switch
		val state = dataManager.areNotificationsEnabled()
		val switch = view.findViewById<Switch>(R.id.switchEventStart)
		switch.isChecked = state
		switch.setOnClickListener {
			dataManager.setNotificationsEnabledState(switch.isChecked)
		}
		val data = dataManager.getAllFoodEvents()
		val events = ArrayList<FoodEvent>()
		try {
			for (d in data) {
				if (!d.isEmpty()) {
					val fe = FoodEvent(d)
					if (fe.timestamp + (fe.duration * 60000) > System.currentTimeMillis())
						events.add(fe)
					else //Remove expired events from memory
						dataManager.removeFoodEvent(fe)
				}
			}
		} catch (e: Throwable) {
			//data is corrupted, delete it all
			dataManager.clearAllFoodEvents()
		}
		//TODO: Refresh button? Or detect changes to shared preferences?
		events.sort()
		val list = view.findViewById<ListView>(R.id.listViewFoodEvents)
		list.emptyView = view.findViewById(R.id.empty)
		list.adapter = FoodEventAdapter(events, view.context, inflater, dataManager)

		return view
	}

}
